source/extensions/http/cache/file_system_http_cache/file_system_http_cache.cc

```diff
@@ -1,18 +1,32 @@
 #include "source/extensions/http/cache/file_system_http_cache/file_system_http_cache.h"
 
+#include <chrono>
+
+#include "source/common/api/os_sys_calls_impl.h"
 #include "source/common/filesystem/directory.h"
 #include "source/common/http/header_map_impl.h"
+#include "source/extensions/http/cache/file_system_http_cache/cache_eviction_thread.h"
 #include "source/extensions/http/cache/file_system_http_cache/cache_file_fixed_block.h"
 #include "source/extensions/http/cache/file_system_http_cache/cache_file_header_proto_util.h"
 #include "source/extensions/http/cache/file_system_http_cache/insert_context.h"
 #include "source/extensions/http/cache/file_system_http_cache/lookup_context.h"
 #include "source/extensions/http/cache/file_system_http_cache/stats.h"
 
+static constexpr Envoy::SystemTime timespecToChrono(const struct timespec& t) {
+  return Envoy::SystemTime{std::chrono::duration_cast<std::chrono::system_clock::duration>(
+      std::chrono::seconds{t.tv_sec} + std::chrono::nanoseconds{t.tv_nsec})};
```

Comment:
Feels like this utility should go into source/common/common/utility.cc

Reply:
Done. Included a bit of a shabby test because there's not really a good test for such a simple thing. (Made some effort to make the test not just a duplicate of the code!)